### PR TITLE
#1074 bugfix (wrong login data/suspicious attempt)

### PR DESCRIPTION
**Pull request by converge**
* when suspicious account is detected, send security code to users email, and ask it in console.

### CODE REVIEW

**Comment by timgrossmann** [comment]
@converge This seems like a good quick fix for the problem.
The question is, if Instagram will display that suspicious login attempt every time the bot starts or only once.

**Comment by converge** [comment]
hey Tim, I tried here with a friend account, it asked me for security code two times, after that I can login without the security code.

**Comment by Naramsim** [comment]
In my case, it keeps asking and asking and asking.....

**Comment by nilkesede** [comment]
~doesn't work for me :( 
it says `Unable to locate email or phone button`~

I actually made it work changing some elements xpath. 
It keep asking for the verification code every time; maybe #1064 solve it

**Comment by converge** [comment]
@nilkesede sounds we didnt found any button to select at the suspicious attempt screen. The expected buttons are (select email or select phone) verification.

Could you take a screenshot of the page that raises that message "Unable to locate email or phone button" ?

**Comment by nilkesede** [comment]
this worked for me https://github.com/nilkesede/InstaPy/commit/172e47869a3771981fdd0d5a726b254e977df17c

**Comment by converge** [comment]
I think we have a final solution for this.

**Comment by Namaslayer** [comment]
@converge I'm willing to help! I got scared that IG would delete my accounts, so I deleted the servers thinking the IP addresses were dangerous to keep using. But I have a few new servers up and running. I'll have to see if i run into the same issues again.

Is there something else I can do in the mean time?

**Comment by converge** [comment]
@Namaslayer for now.. I think we can just test and improve it.

**Comment by thomsonza** [comment]
@converge I have tested this on my Ubuntu server. Get the following error message on running with the `bypass_suspicious_attempt` in InstaPy session initiation:

```
  Traceback (most recent call last):
  File "instagramtest.py", line 19, in <module>
    session = InstaPy(username=insta_username, password=insta_password,nogui=True,bypass_suspicious_attempt=True)
TypeError: __init__() got an unexpected keyword argument 'bypass_suspicious_attempt'
```
I then modified the login_util.py so that the `bypass_suspicious_attempt = True` which then works and displays the following:
```
INFO - Session started - 2017-12-20 13:05:21
Cookie file not found, creating cookie...
Instagram detected an unusual login attempt
A security code wast sent to your Email: s*******a@gmail.com
Type the security code here: 
```
However, I don't receive an email from Instagram.

**Comment by converge** [comment]
hey @thomsonza , the bypass_suspicious_attempt is not needed anymore, thanks to @sionking I'm using the cookie approach and if the cookie file exist, it will not try to bypass the suspicious attempt page.

so you can just need the actual:
```session = InstaPy(username=insta_username, password=insta_password)```

about the security code, Instagram delays it sometimes, just try again.

When I was coding, 1 in 20 login attempts, IG shows up this modal:
Imgur
I'm not able to close this modal screen because I'm not able to catch it, so it will raise some kind of Element not found exception, if it happen, you just need to wait some minutes (3 for example) and try again (maybe 2 or 3 times).

**Comment by Basti922** [comment]
Well didn#t work for me:

`root@v22017064714750533:~/InstaPy# python3 sh.py
Wrong login data!
set_automated_followed_pool error
 [Errno 2] No such file or directory: './logs/sebastianheye_followedPool.csv'
Warning: There are 0 people to unfollow , stopping unfollow_users

Session ended
`

**Review by sebmer-com — COMMENTED** [review]
tested on VPS didn#t work :(

**Comment by converge** [comment]
@Basti922 it doesn't seams you're using the PR

**Comment by thomsonza** [comment]
@converge thanks, I got it working again. The cookie file seems to work.

**Comment by converge** [comment]
hey Tim, I think we're ready to merge it. What do you think ?

**Comment by timgrossmann** [comment]
@converge Yes sounds good! I'll merge it. 
Let's see if there will be any big problems with this, looks like a lot of people are able to bypass the login problem with it :+1: 

Thanks everyone for putting that much work into it and contributing!

**Comment by Namaslayer** [comment]
@timgrossmann @converge I just started up a fresh server and fresh version of InstaPy after the merge, and now I'm getting this error with each script.

screen shot 2017-12-21 at 7 18 36 pm


Any ideas on what I can do to fix this?

**Comment by sebmer-com** [comment]
@Namaslayer having the same issue. I kept the old_InstaPy files they work great (again after i shutdown my VPS). So i guess somehow Instagram has two authorization ways. One with Email Or Phone button and the other with this authorization, where i used to klick "that was me".  
whatsapp image 2017-12-07 at 18 32 24

**Comment by FreakyBoy** [comment]
@Basti922 are your sure you using the PR ? 
Since I changed my branch and using the PR I don't have this like of issue anymore 

EDIT: ohh I see the PR was marge to the master :) so you can pull now the new master and then it must work 👍

**Comment by sebmer-com** [comment]
@FreakyBoy 100% sure.

i git cloned https://github.com/timgrossmann/InstaPy/ 

and set new chromedriver annd all.. running it on a totally fresh system.  Getting the error:

CRITICAL - Wrong login data!  
(because)
Unable to locate email or phone button (as mentioned in #1123 )

File: instapy/login_util.py [hunk 1]
```diff
@@ -46,11 +64,77 @@ def login_user(browser, username, password, switch_language=True):
     # update server calls
     update_activity()
 
+    if bypass_suspicious_attempt is True:
+
```
Comment (nilkesede) [inline instapy/login_util.py:68]: @converge you may have to add a `sleep` here to fix  #1123

**Comment by DEno12** [comment]
+1 @Basti922 same problem.

**Comment by Namaslayer** [comment]
@DEno12 @Basti922 Yeah same here. I never got the email or phone option. Always "this was me" or "this wasn't me"

File: instapy/login_util.py (hunk 1 above)
Comment (converge) [reply instapy/login_util.py:68]: @nilkesede why do you say ?

**Comment by converge** [comment]
I need your help to test #1125

**Comment by sebmer-com** [comment]
Well as mentioned before I am running it with the old version on vps now again with super low frequency after some days of a break... This works.... Maybe it was never a "problem" and IG just stores the IP adresses and we (users) screwd.

File: instapy/login_util.py (hunk 1 above)
Comment (nilkesede) [reply instapy/login_util.py:68]: after click on the login button, it needs a little time to load the suspicious login page

**Pull request merged**